### ISSUE TITLE
Contributing a EKS ServiceAccount wrapper IAM module

### ISSUE THREAD

**Issue by chancez** [terraform-aws-modules/terraform-aws-iam#51]
I saw that #37 just added the OIDC module which is awesome. I had a similar module which is now based on this which is great. I was wondering if it would make sense to add another small wrapper around the `iam-assumable-role-with-oidc` module which handles setting values correctly for EKS service accounts. For what it's worth, my current module looks like this:

```
module "eks_service_account_iam_role" {
  source  = "terraform-aws-modules/iam/aws//modules/iam-assumable-role-with-oidc"
  version = "~> 2.0"

  create_role = true
  role_name   = var.role_name
  tags = {
    role = var.role_name
  }
  provider_url                  = "${replace(var.cluster_oidc_url, "https://", "")}"
  oidc_fully_qualified_subjects = ["system:serviceaccount:${var.namespace}:${var.service_account}"]
  role_policy_arns              = var.policy_arns
}
```

I made a PR to handle removing the `https://` prefix in #50, but I was wondering if it would seem useful enough to have a small wrapper which a namespace and service account and maps to the above example (and/or supporting a list of namespace/service account for oidc_fully_qualified_subjects). It's not super complicated so I can see this being something that doesn't need to live in this module, but I figured I would ask.

**Comment by max-rocket-internet** [comment]
How would this wrapper look when implemented or used? How would it shorted or simplify your example there?

**Comment by chancez** [comment]
Ah I can't believe I forgot that. Something like this:

```
module "loki_iam" {
  source    = "terraform-aws-modules/iam/aws//modules/iam-assumable-role-with-oidc/eks_service_account"
  role_name = "eks-role-${module.eks_staging.cluster_name}-loki_logs"

  namespace       = var.loki_namespace
  service_account = var.loki_service_account_name
  policy_arns     = [aws_iam_policy.loki_logs.arn]

  cluster_oidc_url = module.eks_staging.cluster_oidc.url
}
```

**Comment by max-rocket-internet** [comment]
50/50 TBH. It's a little bit tidier but doesn't save much 😅 I'll defer to a maintainer of this repo.

**Comment by bryantbiggs** [comment]
closed by #179
